com.unity.testing.hdrp/TestRunner/HDRP_GraphicTestRunner.cs

```diff
@@ -62,8 +73,27 @@ public IEnumerator Run(GraphicsTestCase testCase)
             yield return new WaitForEndOfFrame();
         }
 
-        // Reset temporal effects on hdCamera
-        HDCamera.GetOrCreate(camera).Reset();
+        if (settings.waitForFrameCountMultiple)
+        {
+            // Get HDRP instance
+            var hdrp = RenderPipelineManager.currentPipeline as HDRenderPipeline;
+
+            // Standard Test
+            if (settings.ImageComparisonSettings.UseBackBuffer) // Using Backbuffer
+            {
+                // When we capture from the back buffer, there is no requirement of compensation frames
+                while (((hdCamera.cameraFrameCount) % (uint)settings.frameCountMultiple) != 0) yield return new WaitForEndOfFrame();
+            }
+            else
+            {
+                // Given that we will render two frames, we need to compensate for them in the waiting
+                // After this line, the next frame will be frame 0.
+                while (((hdCamera.cameraFrameCount + 2) % (uint)settings.frameCountMultiple) != 0) yield return new WaitForEndOfFrame();
+            }
+        }
+
+        // Force clear all the history buffers
+        hdCamera.RequestClearHistoryBuffers();
```

Comment:
I'm not sure this is the right way to do that.
With this code, we clear all history buffers to black regardless of what the correct default value is. It makes things more consistent but probably wrong in many cases.
A lot of the render loop code relying on history buffers actually checks for first frame or existence of the history buffer to make special cases for when history is not correct. With this way of doing things, this won't work.
I suggest that instead of clearing to black we just dealloc all history buffers. This should mimic better what happens when we actually creates a new camera. Pinging @anisunity as I think it was you who did this?

Reply:
Indeed for all ray tracing buffers black is the default value so it works, but if there is other buffers that require specific values it won't work. deallocating would work yes.